ds-caselaw-ingester/lambda_function.py

```diff
@@ -116,27 +116,45 @@ def extract_lambda_versions(versions: List[Dict[str, str]]) -> List[Tuple[str, s
 
 
 def store_metadata(uri: str, metadata: dict) -> None:
-    tdr_metadata = metadata["parameters"]["TDR"]
-
-    # Store source information
-    api_client.set_property(
-        uri, name="source-organisation", value=tdr_metadata["Source-Organization"]
-    )
-    api_client.set_property(uri, name="source-name", value=tdr_metadata["Contact-Name"])
-    api_client.set_property(
-        uri, name="source-email", value=tdr_metadata["Contact-Email"]
-    )
-    # Store TDR data
-    api_client.set_property(
-        uri,
-        name="transfer-consignment-reference",
-        value=tdr_metadata["Internal-Sender-Identifier"],
-    )
-    api_client.set_property(
-        uri,
-        name="transfer-received-at",
-        value=tdr_metadata["Consignment-Completed-Datetime"],
-    )
+    try:
+        tdr_metadata = metadata["parameters"]["TDR"]
+        # Store source information
+        api_client.set_property(
+            uri, name="source-organisation", value=tdr_metadata["Source-Organization"]
+        )
+        api_client.set_property(
+            uri, name="source-name", value=tdr_metadata["Contact-Name"]
+        )
+        api_client.set_property(
+            uri, name="source-email", value=tdr_metadata["Contact-Email"]
+        )
+        # Store TDR data
+        api_client.set_property(
+            uri,
+            name="transfer-consignment-reference",
+            value=tdr_metadata["Internal-Sender-Identifier"],
+        )
+        api_client.set_property(
+            uri,
+            name="transfer-received-at",
+            value=tdr_metadata["Consignment-Completed-Datetime"],
+        )
+    except KeyError:
```

Comment:
Can we be more explicit about this? Perhaps we switch if a specific field does/does not exist (e.g. if there's not a `TDR` field, or a `fakeTDR` exists?)

Alternatively,

```
try:
        tdr_metadata = metadata["parameters"]["TDR"]
except KeyError:
       ...
else:
      normal_route
```
could narrow the failing line down to one?